### PR TITLE
New package: cc65-2.18

### PR DESCRIPTION
**Pull request by NateDogg1232**
I tried using the gnu-makefile build style, but it would not compile using that, so I made my own do_build and do_install. I do hope this is okay.

Otherwise, it all seems good to me!

### CODE REVIEW

**Comment by NateDogg1232** [comment]
What can I do to solve those architecture issues?

**Comment by Chocimier** [comment]
Following cross-builds, but then Makefile tries to run just-compiled binary. I would ask upstream to don't run as part of build phase and to use toolchain from env.

```
build_style=gnu-makefile
make_use_env=yes
pre_build() {
	vsed -i src/Makefile -e 's/CC =/CC ?=/' -e 's/AR =/AR ?=/'
```

File: srcpkgs/cc65/template [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+revision=1
+short_desc="Freeware C compiler for 6502 based systems"
+maintainer="Nathaniel Barragan <nathanielbarragan@gmail.com>"
+license="Zlib"
+homepage="https://cc65.github.io/"
+distfiles="https://github.com/cc65/cc65/archive/V${version}.tar.gz"
+checksum=d14a22fb87c7bcbecd8a83d5362d5d317b19c6ce2433421f2512f28293a6eaab
+
+do_build() {
+	make PREFIX=/usr
```
Comment (pullmoll) [inline srcpkgs/cc65/template]: You should use `make ${makejobs} PREFIX=/usr` to enable parallel builds, unless that would fail with this package.

Comment (pullmoll) [reply srcpkgs/cc65/template]: Oops, didn't see the suggested fix to build with gnu-makefile style.

**Comment by NateDogg1232** [comment]
Just to make sure people don't think it's dead, I am still working on this. I'm just trying to make sure things work on multiple architectures before comitting and everything.

**Comment by NateDogg1232** [comment]
I found a solution. Upstream, in their Travis builds for Windows, will build the binary, then build the libraries that it needs to build, then will build the cross-binary. I am emulating that in my build file now. It is not pretty, but it is something.

**Comment by NateDogg1232** [comment]
I have decided that cross building is just simply impossible with this package.

**Comment by Chocimier** [comment]
Fine, hopefully it will be easier with future versions. Use `build_style=gnu-makefile`, set `nocross="Tries to run compiled code"`, simplify template.

**Comment by NateDogg1232** [comment]
Don't forget to xlint before pushing, haha. That was my bad, I'm sorry for the spam.

**Comment by NateDogg1232** [comment]
Okay. Maybe this time will be the right commit. I kept forgetting to add the file to the stage before committing.

**Comment by NateDogg1232** [comment]
I am not quite understanding why the musl build is stalling. I am running the musl version of void on my machine and I am not getting a stall in the build.

**Comment by Chocimier** [comment]
Here builds well too.

**Pull request merged**